librtt/Display/Rtt_ShapePath.cpp

```diff
@@ -410,6 +410,58 @@ ShapePath::GetTextureExtents( const ArrayVertex2& texVertices ) const
 	return extents;
 }
 
+bool
+ShapePath::SetFillVertexColor( U32 index, U32 color )
+{
+	fFillGeometry->AttachPerVertexColors( &fFillSource.Colors(), GetFillVertexCount() );
```

Comment:
I'm not sure I like this path... This pretty much ensures that tessellation happens twice always. Do we need to `GetFillVertexCount` here always? Do you think there's a way to attach vertex colors without knowing number of vertices in advance, then just ignoring "extra" colors?

Reply:
Basically, it is fine to ask to tesselate if we explicitly request number of vertices. But if we know it in advance, we can skip tessellation, like for mesh or quad. Not sure it is an issue, as it would have to happen eventually anyways, we're just removing laziness.
what do you think? I'm not asking to make changes here (yet) just for a discussion.

Reply:
Thanks!

I would love a better solution. In my original implementation with meshes only, I was able to ask the (mesh-specific) tessellator for its `Length()`. For most of the other shapes and lines, adding this probably wouldn't be so hard, as say a `virtual`. With circles (and maybe rounded rects) I wasn't quite as confident, or at least expected some pain. 😄

Reply:
Whoops, saw you already mentioned meshes.

In retrying my sample, I see that path really does get hit hard, especially with the circle! I'm looking over the tessellators and it doesn't look too bad to add a vertex count method. In particular, all the strokes will be (# points + 1) * 2. Even those rounded ones don't look so bad, now.

I'll take a stab at this.